packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -562,6 +649,33 @@ class TextPainter {
     _layoutTemplate = null; // Shouldn't really matter, but for strict correctness...
   }
 
+  static double _computePaintOffsetFraction(TextAlign textAlign, TextDirection textDirection) {
+    switch (textAlign) {
+      case TextAlign.left:
+        return 0;
+      case TextAlign.right:
+        return 1;
+      case TextAlign.center:
+        return 0.5;
+      case TextAlign.start:
+      // eot is also considered a hard break.
+      case TextAlign.justify:
+        switch (textDirection) {
+          case TextDirection.rtl:
+            return 1;
+          case TextDirection.ltr:
+            return 0;
+        }
+      case TextAlign.end:
+        switch (textDirection) {
+          case TextDirection.rtl:
+            return 0;
+          case TextDirection.ltr:
+            return 1;
+        }
+    }
```

Comment:
for clarity it might be good to say `0.0` and `1.0` instead of `0` and `1` when they're doubles